### ISSUE TITLE
Operator/Operand overload checking issue 

### ISSUE THREAD

**Issue by tinganho** [Microsoft/TypeScript#7408]
TypeScript should not give an error on the following case:

```ts
let diff = new Date() - new Date();
```
Because the `Date` objects has a `valueOf` method that will convert the `Date` object to a number value during arithmetic operations. Or any expression that expects a primitive type.

Source:
http://stackoverflow.com/questions/24764137/why-is-it-possible-to-subtract-date-objects-in-javascript-is-there-any-form-of

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/valueOf

**Comment by tinganho** [comment]
Oops, already discussed in https://github.com/Microsoft/TypeScript/issues/5337
